content/install-user-css.js

```diff
@@ -0,0 +1,284 @@
+    });
+}
+
+function runtimeSend(request) {
+  return new Promise((resolve, reject) => {
+    chrome.runtime.sendMessage(request, result => {
+      if (result.status === 'error') {
+        reject(result.error);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+function getAppliesTo(style) {
+  function *_gen() {
+    for (const section of style.sections) {
+      for (const type of ['urls', 'urlPrefixes', 'domains', 'regexps']) {
+        if (section[type]) {
+          yield *section[type];
+        }
+      }
+    }
+  }
+  const result = [..._gen()];
+  if (!result.length) {
+    result.push(chrome.i18n.getMessage('appliesToEverything'));
+  }
+  return result;
+}
+
+function initInstallPage({style, dup}, sourceLoader) {
+  return pendingResource.then(() => {
+    const versionTest = dup && semverCompare(style.version, dup.version);
+    document.body.textContent = '';
+    document.body.appendChild(buildPage());
+
+    if (versionTest < 0) {
+      $('.actions').parentNode.insertBefore(
+        $element({className: 'warning', textContent: t('versionInvalidOlder')}),
+        $('.actions')
+      );
+    }
+    $('.code').textContent = style.source;
+    $('button.install').onclick = () => {
+      if (dup) {
+        if (confirm(chrome.i18n.getMessage('styleInstallOverwrite', [style.name, dup.version, style.version]))) {
+          install(style);
+        }
+      } else if (confirm(chrome.i18n.getMessage('styleInstall', [style.name]))) {
+        install(style);
+      }
+    };
+
+    if (location.protocol === 'file:') {
+      initLiveReload(sourceLoader);
+    }
+
+    function buildPage() {
+      return $element({className: 'container', appendChild: [
+        $element({className: 'header', appendChild: [
+          $element({tag: 'h1', appendChild: [
+            style.name,
+            $element({tag: 'small', className: 'meta-version', textContent: style.version})
+          ]}),
+          $element({tag: 'p', textContent: style.description}),
+          $element({tag: 'h3', textContent: t('author')}),
+          style.author,
+          $element({tag: 'h3', textContent: t('license')}),
+          style.license,
+          $element({tag: 'h3', textContent: t('appliesLabel')}),
+          $element({tag: 'ul', appendChild: getAppliesTo(style).map(
+            pattern => $element({tag: 'li', textContent: pattern})
+          )}),
+          $element({className: 'actions', appendChild: [
+            $element({tag: 'button', className: 'install', textContent: installButtonLabel()})
+          ]}),
+          $element({className: 'external', appendChild: [
+            externalLink('externalHomepage', style.url),
+            externalLink('externalSupport', style.support)
+          ]})
+        ]}),
+        $element({className: 'main', appendChild: [
+          $element({className: 'code'})
+        ]})
+      ]});
+    }
+
+    function externalLink(name, url) {
```

Comment:
This is very similar to the `makeLink` function in the `lint.js` file. We could make this DRY and move it to the `dom.js` file.